### PR TITLE
Create corpus model

### PR DESCRIPTION
**Pull request by yaroluchko**
This PR creates the basic corpus model that we can work on and add functionality to.

### CODE REVIEW

File: backend/app/models.py [hunk 1]
```diff
@@ -282,6 +292,7 @@ class Document(models.Model):
     tokenized_text = models.JSONField(null=True, blank=True, default=None)
     word_count_counter = models.JSONField(null=True, blank=True, default=dict)
     part_of_speech_tags = models.JSONField(null=True, blank=True, default=list)
+    corpuses = models.ManyToManyField(Corpus)
```
Comment (sharabhuiyan) [inline backend/app/models.py]: I really like this! This is a really neat way to keep track of documents in Corpora.

File: backend/app/tests.py [hunk 2]
```diff
@@ -119,10 +119,119 @@ class MainTests(TestCase):
...
+        self.assertEqual(doc_1.get_count_of_words(['do', 'as', 'you']), Counter({'do': 2, 'as': 2, 'you': 1}))
+        self.assertEqual(doc_1.get_count_of_words(['ThisWordIsNotThere', 'well']), Counter({'ThisWordIsNotThere': 0, 'well': 0}))
+
+    def test_words_associated(self):
+        doc_1 = Document(text="""She took a lighter out of her purse and handed it over to him.
+                                 He lit his cigarette and took a deep drag from it, and then
+                                 began his speech which ended in a proposal. Her tears drowned the ring.
+                                 TBH i know nothing about this story.""")
+        doc_1.save()
+        self.assertEqual(doc_1.words_associated('his'), Counter({'cigarette': 1, 'speech': 1}))
```
Comment (sharabhuiyan) [inline backend/app/tests.py]: Also consider making a test case with "her" because it also appears a few times in this document.

**Review by sharabhuiyan — APPROVED** [review]
Good job! The testing seems fine and I really liked how you guys utilized many to many relationships.

**Review by MBJean — CHANGES_REQUESTED** [review]
This pretty much looks good to me, thanks for your review, Shara! One thing, though: are the changes in `texts.py` intentional? They all concern the `Document` model, and look duplicative of what's in the `DocumentTestCase` class above.

**Comment by phuang00** [comment]
We fixed the duplicate test cases in `tests.py` and we added test cases to ensure that we can properly associate the `corpus` and `document` models.

**Comment by YifanWang0** [comment]
We also created a Corpus serializer.

**Review by MBJean — APPROVED** [review]
This generally looks good, thanks! I have two pieces of feedback below. Let me know what you think, and if you agree, they should be easy changes to make, so I'll leave my approval here.

1. We'll want to manually add the plural `corpora` to our `Corpus` model. You can do so by following the guidelines outlined [here](https://books.agiliq.com/projects/django-admin-cookbook/en/latest/plural_text.html). Otherwise, we'll end up in a weird situation, as we already are in with `PronounSeries`, where we get a silly plural name `Corpuss` and `PronounSeriess`, respectively. See this screencap of the admin page without manually specifying the plural:

Screen Shot 2021-07-10 at 3 01 57 PM

2. I'm thinking that the `ManyToMany` field should be specified on the `Corpus` model, not the `Document` model. That's my impulse because I'm imaging that a user is going to create a `Corpus` in the UI, then pick which `Document`s to add via a dropdown or something. In other words, the user will be working in the context of a `Corpus`, not a `Document`, when adding `Document`s to a `Corpus`, and so adding the `ManyToMany` field to `Corpus` more accurately reflects that behavior. [An old version of the Django docs](https://docs.djangoproject.com/en/1.8/topics/db/models/#many-to-many-relationships) recommends as such, as well.

**Comment by phuang00** [comment]
> 2. I'm thinking that the `ManyToMany` field should be specified on the `Corpus` model, not the `Document` model. That's my impulse because I'm imaging that a user is going to create a `Corpus` in the UI, then pick which `Document`s to add via a dropdown or something. In other words, the user will be working in the context of a `Corpus`, not a `Document`, when adding `Document`s to a `Corpus`, and so adding the `ManyToMany` field to `Corpus` more accurately reflects that behavior. [An old version of the Django docs](https://docs.djangoproject.com/en/1.8/topics/db/models/#many-to-many-relationships) recommends as such, as well.

@MBJean 
We were also looking at the Django docs when we were writing the `ManyToMany` field and the [docs](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/) had the opposite convention so we followed that even though it seemed counterintuitive. Which Django docs should we follow?

**Comment by MBJean** [comment]
Good call out. I think the doc you linked is perhaps just not particularly specific. In the general models docs ([here's](https://docs.djangoproject.com/en/3.2/topics/db/models/#many-to-many-relationships) a more updated link than what I first posted above), the example of Pizza and Topping models has the ManyToMany field on Pizza. That seems to correspond to our use case here. What do you think?

**Comment by phuang00** [comment]
Yeah, I think the ManyToMany field for Pizza and Topping makes more sense intuitively and match with what we're doing. We'll update the field!

**Pull request merged**